src/MaintenanceHistoryAPI/EventHandlerHistory.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Diagnostics;
+
+namespace Pitstop.MaintenanceHistoryAPI;
+
+public class EventHandlerHistory(IMessageHandler messageHandler) : IHostedService, IMessageHandlerCallback
+{
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        var activitySource = new ActivitySource("Pitstop.MaintenanceHistoryAPI.EventHandlerHistory");
+        messageHandler.Start(this, activitySource);
+        
+        Console.WriteLine("EventHandlerHistory started.");
+        
+        await Task.CompletedTask;
+        
+        Console.WriteLine("EventHandlerHistory started.");
+    }
+
+    public Task StopAsync(CancellationToken cancellationToken)
+    {
+        messageHandler.Stop();
+        return Task.CompletedTask;
+    }
+
+    public async Task<bool> HandleMessageAsync(string messageType, string message)
+    {
+        try
+        {
+            var messageObject = MessageSerializer.Deserialize(message);
+            switch (messageType)
+            {
+                case "MaintenanceHistory":
+                    await HandleAsync(messageObject.ToObject<MaintenanceHistory>());
+                    break;
+            }
+        }
+        catch (Exception ex)
+        {
+            Log.Error(ex, $"Error while handling {messageType} event.");
```

Comment:
Is Log.Error zelfde als `throw new Error`? Of is dat geen C#?

Reply:
Log.<watever> is voor de logging tool, dus logt een error naar de logservice